File: backend/app/usecases/user_service.py
from typing import List
from ..domain.models import User
from ..infrastructure.repositories import UserRepository

class UserService:
    def __init__(self, repo: UserRepository):
        self.repo = repo

    def list_users(self) -> List[User]:
        return self.repo.all()

    def create_user(self, name: str) -> User:
        return self.repo.create(name)

File: backend/src/usecases/userService.ts
import { User } from '../domain/user'
import { Database } from '../infrastructure/db'

export class UserService {
  constructor(private db: Database) {}

  async listUsers(): Promise<User[]> {
    return this.db.getUsers()
  }

  async createUser(name: string): Promise<User> {
    return this.db.createUser(name)
  }
}
